File: src/commands/help.js
module.exports = {
    name: "help",
    description: "prints the help message for all the commands",
    usage: "<prefix>pingchallenges",
    private: false,
    execute(message) {
        const commands = message.client.textCommands;

        const data = [];

        if (!commands) {
            return message.channel.send("There are no commands available.");
        }

        data.push("**Here's a list of all my commands:**");

        data.push("\n```"); // creates a code block for the commands to sit in
        commands.forEach((command) => {
            if (command.private) return;

            data.push(`Name: ${command.name}`);
            data.push(`Description: ${command.description}`);
            data.push(`Usage: ${command.usage}`);
            data.push(`\n`)
        });
        data.push("```");

        message.channel.send(data.join("\n"));
    }
}

File: src/commands/help.rs
// help.rs
use serenity::all::CreateMessage;
use serenity::builder;
use serenity::model::channel::Message;
use serenity::prelude::Context;

const COMMANDS_AND_DESCRIPTIONS: [(&str, &str); 3] = [
    ("ping", "Responds with 'Pong!'"),
    ("paste", "Copies a code block to the DPaste"),
    ("help", "Displays this message"),
];

pub async fn execute(ctx: Context, msg: &Message, _args: Vec<&str>) -> Result<(), Box<dyn std::error::Error>> {
    let prefix = std::env::var("PREFIX")?;
    let usage_message = format!("Usage: **{}**<command> <arguments*>", prefix);

    let embed_fields = COMMANDS_AND_DESCRIPTIONS.iter().map(|(command, description)| {
        (command.to_string(), description.to_string(), false)
    }).collect::<Vec<_>>();

    let embed = builder::CreateEmbed::default()
        .title("Here is some help!")
        .description(usage_message)
        .fields(embed_fields)
        .color(0xFFB6C1);

    let message = CreateMessage::new()
        .embed(embed);

    msg.channel_id
        .send_message(&ctx.http, message)
        .await?;
    Ok(())
}